src/screens/ConfirmEmail.js

```diff
@@ -0,0 +1,190 @@
+    fieldFour: '',
+  });
+
+  useEffect(() => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  }, []);
+
+  const handleOnChange = ({number, id}) => {
+    setFormData({
+      ...formData,
+      [id]: number,
+    });
+  };
+
+  const resetForm = () => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  };
+
+  const checkCodeConfirmation = () => {
+    if (
+      formData.fieldOne === '' &&
+      formData.fieldTwo === '' &&
+      formData.fieldThree === '' &&
+      formData.fieldFour === ''
+    ) {
+      Alert.alert('type all the code');
+    } else {
+      let userCodeConfirmation = [
+        formData.fieldOne,
+        formData.fieldTwo,
+        formData.fieldThree,
+        formData.fieldFour,
+      ];
+      for (let i = 0; i < 4; i++) {
+        if (codeConfirmation[i] === userCodeConfirmation[i]) {
+          auxiliarCounter++;
+        }
+      }
+      if (auxiliarCounter === 4) {
+        Alert.alert('Success');
+      } else {
+        Alert.alert('Your code is wrong');
+      }
+    }
+  };
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        <Text style={styles.title}>Confirmación de correo</Text>
+        <Text style={styles.textInstruction}>
+          Por favor ingresa el código enviado a{' '}
+          <Text style={[styles.textInstruction, styles.userEmail]}>
+            {usersEmail}
+          </Text>
+        </Text>
+        <View style={styles.inputsContainer}>
+          <CodeConfirmTxtInput
+            id="fieldOne"
+            value={formData.fieldOne}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldTwo"
+            value={formData.fieldTwo}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldThree"
+            value={formData.fieldThree}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldFour"
+            value={formData.fieldFour}
+            onChangeText={handleOnChange}
+          />
+        </View>
+        <View style={styles.requireNewCodeContainer}>
+          <Text style={styles.newCodeInstructions}>
+            ¿No recibiste el código? {'\n'} Revisa tu carpeta de correos no
+            deseados
+          </Text>
+          <TouchableOpacity
+            style={styles.sendNewCodeBtn}
+            onPress={() => resetForm()}>
+            <Text style={styles.sendNewCodeTxt}>Enviar código de nuevo</Text>
+          </TouchableOpacity>
+        </View>
+        <TouchableOpacity
+          style={styles.nextBtn}
+          onPress={() => checkCodeConfirmation()}>
+          <Text style={styles.nextMessageTxt}>{NEXT_MESSAGE}</Text>
+        </TouchableOpacity>
+      </ScrollView>
+    </SafeAreaView>
+  );
+};
+
+export default ConfirmEmail;
```

Comment:
is this component `ConfirmEmail` completely reusable? its children's can be splited in other reusable components.